lib/src/realm_class.dart

```diff
@@ -156,10 +156,12 @@ class Realm {
 
     final handle = metadata.class_.primaryKey == null
         ? realmCore.createRealmObject(this, metadata.class_.key)
-        : realmCore.createRealmObjectWithPrimaryKey(this, metadata.class_.key, object.accessor.get(object, metadata.class_.primaryKey!)!);
+        : update
+            ? realmCore.getOrCreateRealmObjectWithPrimaryKey(this, metadata.class_.key, object.accessor.get(object, metadata.class_.primaryKey!)!)
```

Comment:
I think this is really a bad practice (at least in SQL it was) and should be done at the Core level. I will add a discussion point about this.